src/mame/drivers/tetrisp2.cpp

```diff
@@ -1860,53 +1839,174 @@ TIMER_DEVICE_CALLBACK_MEMBER(stepstag_state::field_cb)
 	// irq 4 is definitely a 30 Hz-ish here as well,
 	// except we have a multi-screen arrangement setup and no way to pinpoint source
 	m_subcpu->set_input_line(4, HOLD_LINE);
+	stepstag_state::simulate_pc();	// any better place to put it into schedule?
 }
 
 void stepstag_state::setup_non_sysctrl_screen(machine_config &config, screen_device *screen, const XTAL xtal)
 {
 	// TODO: unknown clock and parameters
-	// assume there's a 42.954 MHz/6 like nndmseal to compensate the higher res
-	screen->set_raw(xtal/6, 455, 0, 352, 262, 0, 240);
+	screen->set_raw(xtal, 429, 0, 352, 525, 0, 240);	// should be 0-479 interlaced
 }
 
+void stepstag_state::simulate_pc()
+{
+	/*command list(stepstag):
+	$010x - start mpeg playing?
+	$02xx - ?
+	$03xx - open mpeg file set
+		$0311 jaleco logo animation
+		$0319 demo L/R bg
+	$04xx - ?
+	$07xx - load sound files to particular addresses of ymz280b ram
+		$070x stereo game music
+		$071x sound effects
+		$072x sample set for music selection
+		$07ff clear music sample region?
+	*/
+	FILE *inFile;
+	unsigned int size, position;	// position2
+	const unsigned int total_ram = 0x1000000;	// full 16MB for each ymz280b
+	if(strcmp(machine().system().name,"vjdash") == 0)
```

Comment:
This is really discouraged. Either split `vjdash `to a separate sub-state of `stepstag` (does it have any other state machine difference?) or just delegate to a specific const variable `position_start`.

Reply:
Yeah, this is the kind of thing that can be set in a data member from a config helper function, or at worst a driver init function.  Definitely shouldn't be done like this.

Reply:
The "simulate_pc" code is quite elementary and experimental, only for purpose of development, and share with you all what I've discovered.
Will not include these code this time, or make them as remarks.